src/edu/washington/escience/myria/io/AmazonS3Source.java

```diff
@@ -95,7 +96,12 @@ public AmazonS3Client getS3Client() throws MyriaApiException {
         clientConfig.setMaxErrorRetry(3);
         s3Client = new AmazonS3Client(credentials, clientConfig);
       } catch (ClassNotFoundException e) {
-        throw new MyriaApiException(Status.INTERNAL_SERVER_ERROR, className + " not found ", e);
+        if (className == null) {
```

Comment:
Probably should check for `null` and throw immediately after you assign `className`, since I'm not sure `Class.forName()` would actually throw a `ClassNotFoundException` for a `null` class name, and even if it did, I wouldn't want to assume that behavior would continue since it's not documented AFAICT.